pika/adapters/blocking_connection.py

```diff
@@ -370,14 +377,12 @@ def _process_io_for_connection_setup(self):  # pylint: disable=C0103
 
         :raises AMQPConnectionError: on connection open error
         """
+        self._check_open_error()
+
         self._flush_output(self._opened_result.is_ready,
```

Comment:
This method is small and the new `_check_open_error` method is only used from within this method. The code would be more readable to leave the old logic in place and simply make the call to `_flush_output` conditional on `self._open_error_result.ready`, like this:

```
if not self._open_error_result.ready:
    self._flush_output(self._opened_result.is_ready, ...

if self._open_error_result.ready:
    exception_or_message = self._open_error_result.value.error
    ...
```